Gruppe.A1.End-to-End_Encryption_mit_IPv6_und_integriertem_IPSec.ThushjandanPonnudurai.PascalBaumann.Termpaper.tex

```diff
@@ -62,12 +73,103 @@ \chapter{Theoretical part}
 \subsection{IPv6}
 \label{ssec:IPv6}
+Internet Protocol version 6 (IPv6) is the successor of IPv4. It was specified in 1998. The growth of the Internet led to a need for new alternatives. The problem is that IPv4 cannot provide the needed number of logical addresses. For that reason IPv6 protocol was developed.\parencite[11]{Babatunde2014}
+
+Several important areas were enhanced in the IPv6 protocol:
+\begin{itemize}
+	\item Expanded addressing
+	\item Simplified header format
+	\item Improved extension and option support
+\end{itemize}
+With the expanded address space the issue with the lack of addresses will be eliminated. Also the addressing architecture and the simplified header format are improving the routing efficiency. Fewer packets needs to be processed thanks to the simplified header format. 
+Every IPv6 packet has a fixed header length of 40 octets. But any options can be appended in extension headers after the IPv6 main header. \parencite[106-107,123-124]{Loshin2004}
+
+\subsubsection{IPv6 Addressing}
+\label{sssec:ipv6:addressing}
+There are three types of IPv6 addresses:
+\begin{itemize}
+	\item Unicast. A packet sent to an unicast address is delivered to the interface identified by that address.
+	\item Multicast. A packet sent to a multicast address is delivered to all interfaces identified by that address. This address could be existing on different hosts.
+	\item Anycast. A packet sent to a anycast address is delivered to one of the interfaces identified by that address. This address could be existing on different hosts.
+\end{itemize}
+Broadcast is not existing anymore in IPv6. \parencite[142-143]{Loshin2004}
+
+The length of an IPv6 address is 128-bit. It is usually represented as 8 groups of up to 4 hexadecimal digits that are separated with colons. Each group stands for 16 bit.
+\begin{quotebox}
+	XXXX:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX
+\end{quotebox}
+Here is an example of an IPv6 address.
+\begin{quotebox}
+	2001:db8:1234:ABCD::10
+\end{quotebox}
+IPv6 addresses are usually divided into two equal 64-bit parts. The first 64-bit part identifies a network address and the last 64-bit part the host.\parencite[144-146]{Loshin2004}
+ 
+\subsubsection{IPv6 Headers}
+\label{sssec:ipv6:headers}
+The length of an header in IPv6 is not variable like in IPv4. It has a fixed length of 40 bytes. An IPv4 header can be as short as 20 bytes and as long as 60 bytes due to the IP options. Routers process usually always the options in the IPv4 header. Even if the options aren't relevant for the processing router. This leads to a degradation of the forwarding performance.  With the fixed header length in IPv6, packet handling is more efficient. \parencite[128]{Loshin2004}
```

Comment:
[...] header _, even if these options_ aren't relevant for the processing router.